src/internal/connector/graph_connector_test.go

```diff
@@ -249,6 +250,18 @@ func (suite *GraphConnectorUnitSuite) TestPopulateOwnerIDAndNamesFrom() {
 			expectName: name,
 			expectErr:  require.NoError,
 		},
+		{
+			name:  "site suffix lookup",
```

Comment:
should have a test to show that it consistently selects the same ("correct"?) site when there are multiple sites that have the same suffix

Reply:
Hmm, we're not making any guarantees there (outside of basic correctness) in the first place.  Ordering matters.  The result will be consistent for a given array, but not for a given set of values.  I'm going to punt this to a follow-up PR where I can look at longest/shortest match control more generically.